src/elements/forms/styles.css

```diff
@@ -0,0 +1,26 @@
+:root {
+  --textarea-min-line-count: 1;
+  --textarea-line-count: 3;
+  --form-control-line-height: var(--line-height-base, 1.5);
+}
+
+/* Restore height for the text area */
+textarea {
+  height: calc(
+    2 * var(--form-control-padding-y) + var(--form-control-line-height) *
+      var(--textarea-line-count) * var(--form-control-font-size)
+  );
+  height: auto;
+  min-height: calc(
+    2 * var(--form-control-padding-y) + var(--form-control-line-height) *
+      var(--textarea-min-line-count) * var(--form-control-font-size)
+  );
+  padding-top: calc(var(--form-control-padding-y) - 1px);
+  padding-bottom: calc(var(--form-control-padding-y) - 1px);
+  line-height: var(--line-height-base, 1.5);
+}
+
+select:not([multiple]):not([size]) {
+  background-position: right center;
+  background-origin: content-box;
+}
```

Comment:
This is the tricky part. Select is styled by [applying a triangle icon on the background](https://github.com/csstools/sanitize.css/blob/5bd695fdc7be25221c542810a72c174a33d6fb00/forms.css#L35). The current code shifts this to the content box to skip 1rem (16 pixels) from the right.

The problems are:
1. we should apply our own icon from `@mdi/svg`
2. the icon should be styled and ideally should change the colour with the border colour
3. ideally it should be applied with CSS too what makes it even more tricky
    - the trickiness is that if we apply the icon with CSS, it should be so everywhere
    - we probably should support different colour icons but maybe not, I am not sure yet
    - if the icon is applied with CSS, JS API will be different from proposed in #197 originally
4. the icon and the content should not overlap, so shifting to the content is not correct
    - the padding-right should be expanded in 3 times: `16px` for padding, `16px` for the icon box, `16px` more for the padding between the icon and the actual text box
    - in this case, positioning is not so easy since background aligns [very specifically](https://developer.mozilla.org/en-US/docs/Web/CSS/background-position)

Reply:
> we probably should support different colour icons but maybe not, I am not sure yet

IMHO icon should always be dependent on `currentColor`.


> in this case, positioning is not so easy since background aligns very specifically
Why not to have it as `:before` and `:after` element. Anyway if possible it would be better if icons are not in CSS so a user can use any icon they want.

I would say there is one more problem - **icons should be part of sprite ideally**.


It's a tough task. Thanks for working on it. 🍀 🤞

Reply:
> icon should always be dependent on `currentColor`.

It's not like that. Here it should be `var(--form-control-color)` specifically. The `currentColor` is the value of `color` property in the current element, i.e. it's the text colour, i.e. black, what does not help us.

I could use the background blend mode if it would work for icons but I am afraid not. I will try anyway.

> icons should be part of sprite ideally

Ideally yes. Technically it does not work. CSS is tied to HTML, you can only reference SVG symbols written in the HTML but cannot use external ones.

It's possible to programmatically build a CSS file with all icons in it exposed as classes or variables. It might work but I am not completely sure in the need of this.

> Why not to have it as `:before` and `:after` element.

These pseudo-elements are not available for the `<input>` element. I would do so if it would be possible 🙂

The decision should rely on the way of using the icons. I see a presentational role for it only. It does not require any accessibility because the text label always exist and for valid/invalid statuses we have special reporting too. Icons are complementary only and can be part of the background.